src/api/httpClient.ts

```diff
@@ -41,14 +41,14 @@ function queryParams() {
   }
 }
 
-function getUrl(path: string, options?: Record<string, string>) {
+function getUrl(path: string, options?: Record<string, string | undefined>) {
   const serverUrl = useAppStore.getState().data.url
   const params = new URLSearchParams(queryParams())
 
   if (options) {
     Object.keys(options).forEach((key) => {
       if (options[key] !== undefined) {
-        params.append(key, options[key])
+        params.append(key, options[key]!)
```

Comment:
Yeah, I was a little surprised as well but without it I get this:
```
> tsc && vite build

src/api/httpClient.ts:51:28 - error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.

51         params.append(key, options[key])
                              ~~~~~~~~~~~~


Found 1 error in src/api/httpClient.ts:51

 ELIFECYCLE  Command failed with exit code 2.
```

Reply:
I guess I overestimated TypeScript—my bad, lol. It can be a bit frustrating at times. When checking `options[key]` in an if statement, it gets confused when trying to assign it to `params.append`.

```js
Object.keys(options).forEach((key) => {
    const query = options[key]

    if (query !== undefined) {
      params.append(key, query)
    }
})
```

If you extract the value beforehand, it doesn't get lost.

Reply:
Good to know, I'll keep this in mind for next time!